src/tests/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+
+if(ENABLE_TESTING)
+  foreach(PROG test_elements )
+    ADD_DEFINITIONS(-DBOOST_TEST_DYN_LINK) 
```

Comment:
Will this break linking against static boost test framework libraries?

And please use:
```
target_compile_definitions(unit_${PROG} PRIVATE BOOST_TEST_DYN_LINK)
```
instead.

Reply:
I'm not sure. Are we using static libraries for boost in general, or is it meant to be an option where both are supported? If so is there a variable associated with static vs dynamic linking defined somewhere in the cmake files?

Reply:
I was just curious. The main problem being that distribution don't package static libraries, so let's leave like this for now and wait for complains.